### DISCUSSION TITLE
【教程】教程顺序更改意见征集

### DISCUSSION THREAD

**Discussion by AmazingAng** [AmazingAng/WTF-Solidity#72] [General]
## 入门
| 新顺序 | 旧顺序 | 标题 |
| ----------- | ----------- | ----------- |
| 1 | 1 |三行代码HelloWeb3 |
| 2 | 2 | 数值类型 |
| 3 | 3 | 函数类型 |
| 4 | 18 | 函数输出 |
| 5 | 4 | 变量作用域和数据存储 |
| 6 | 5 | 数组和结构体 |
| 7 | 6 | 映射 |
| 8 | 20 | 变量初始值 |
| 9 | 14 | 常量 |
| 10 | 7 | 控制流 |
| 11 | 8 | 构造函数和修饰器 |
| 12 | 9 | 事件 |
| 13 | 10 | 继承 |
| 14 | 11 | 抽象合约和接口 |
| 15 | 13 | 异常 |

## 进阶
| 新顺序 | 旧顺序 | 标题 |
| ----------- | ----------- | ----------- |
| 16 | 15 |函数重载 |
| 17 | 12 | 库合约 |
| 18 | 30 | Import |
| 19 | 16 | 接收ETH |
| 20 | 17 | 发送ETH |
| 21 | 19 | 调用其他合约 |
| 22 | 21 | Call |
| 23 | 22 | Delegatecall |
| 24 | 23 | 在合约中创建合约 |
| 25 | 24 | Create2 |
| 26 | 25 | 删除合约 |
| 27 | 26 | ABI编码解码 |
| 28 | 27 | Hash |
| 29 | 28 | 函数选择器 |
| 30 | 29 | Try-Catch |

**Comment by aihes** [comment]
牛~

**Comment by HardenV** [comment]
厉害

**Comment by iaskgithub** [comment]
严重关注
